lib/flowcommerce_spree/webhook_service.rb

```diff
@@ -111,6 +111,8 @@ def hook_order_upserted_v2
           end
         end
 
+        attrs_to_update.merge(order.upsert_flow_addresses) if order.complete?
```

Comment:
Use `merge!` instead of `merge` - it is way faster and doesn't initialize a new hash